stingray/crossspectrum.py

```diff
@@ -445,16 +452,17 @@ def _make_segment_spectrum(self, lc1, lc2, segment_size):
         if lc1.tseg != lc2.tseg:
             raise ValueError("Lightcurves do not have same tseg.")
 
-        # number of bins per segment
-        nbins = int(segment_size/lc1.dt)
-        start_ind = 0
-        end_ind = nbins
+        if gti is None:
+            gti = cross_two_gtis(lc1.gti, lc2.gti)
```

Comment:
I did this because the user might want to further filter the data without messing with the GTIs of the single light curves.
For example: The light curve gtis are both [[0, 100]]. I want to filter out the interval [[20,30]] because I suspect something went wrong there or the flux was higher than a certain threshold. In the current way, I can just give my handmade GTI and calculate the cross spectrum. Otherwise, I would have to change the GTI of the light curves, and then calculate the cross spectrum. But the logical step here doesn't sound right to me: I want to play with GTIs during this spectral calculation, I don't necessarily want to modify the GTIs of single light curves.